caddyhttp/basicauth/basicauth.go

```diff
@@ -52,37 +52,39 @@ func (a BasicAuth) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error
 	var protected, isAuthenticated bool
 	var realm string
 
-	for _, rule := range a.Rules {
-		for _, res := range rule.Resources {
-			if !httpserver.Path(r.URL.Path).Matches(res) {
-				continue
+	if r.Method != http.MethodOptions {
```

Comment:
Could we return early if this is the case
```
// Do not check basicauth for OPTIONS
if r.Method == http.MethodOptions {
// Pass-through when no paths match
	return a.Next.ServeHTTP(w, r)
}

for _, rule := range a.Rules {
//etc
```
I think it works better for usability

Reply:
good point 👍 
i changed the check accordingly